src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,98 @@
+package duke;
+
+import java.time.DateTimeException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+
+/**
+ * Represents a Parser responsible for parsing Strings
+ * to generate tasks and commands for the Duke bot.
+ */
+public class Parser {
+
+    /** Format for date and time used when user inputs a task */
+    private static final DateTimeFormatter INPUT_DATE_TIME_FORMAT
+            = DateTimeFormatter.ofPattern("dd-MM-yyyy HHmm");
+
+    /** Format for date and time used when Duke saves a task list */
+    private static final DateTimeFormatter SAVED_DATE_TIME_FORMAT
+            = DateTimeFormatter.ofPattern("MMM dd yyyy h:mm a");
+
+    /**
+     * Parses a line read from the task list from the storage space
+     * and generates a corresponding Task to be added to the live TaskList.
+     *
+     * @param fullTask The line from the .txt file to be parsed
+     * @return The corresponding Task.
+     * @throws DukeException If the line contains an incorrect format.
+     */
+    public static Task parseTask(String fullTask) throws DukeException {
+        String[] splitTask = fullTask.split(" \\| ");
+
+        String type = splitTask[0];
+        String description = splitTask[2];
+        boolean isDone = Integer.parseInt(splitTask[1]) == 1;
+
+        switch (type) {
+        case "T":
+            return new Todo(description, isDone);
+        case "D":
+            LocalDateTime by = LocalDateTime.parse(splitTask[3], SAVED_DATE_TIME_FORMAT);
+            return new Deadline(description, isDone, by.toLocalDate(), by.toLocalTime());
+        case "E":
+            LocalDateTime at = LocalDateTime.parse(splitTask[3], SAVED_DATE_TIME_FORMAT);
+            return new Event(description, isDone, at.toLocalDate(), at.toLocalTime());
+        default:
+            throw new DukeException("Unable to read task.");
```

Comment:
Consider adding a `break;` after each case (or specify a `//Fallthrough`, but here I suppose you don't want to do that)